src/fiber/channel.rs

```diff
@@ -454,26 +451,38 @@ where
                 // If there is a next hop, we should send the AddTlc message to the next hop.
                 // If this is the last hop, we should check the payment hash and amount and then
                 // try to fulfill the payment, find the corresponding payment preimage from payment hash.
-                let mut forward_to_next_hop = false;
                 let mut preimage = None;
-                if let Ok(onion_packet) = OnionPacket::deserialize(&add_tlc.onion_packet) {
-                    if let Some(onion_info) = onion_packet.peek() {
-                        if onion_info.next_hop.is_some() {
-                            forward_to_next_hop = true;
-                        } else {
-                            // check the payment hash and amount
-                            if onion_info.payment_hash != add_tlc.payment_hash
-                                || onion_info.amount != add_tlc.amount
-                            {
-                                return Err(ProcessingChannelError::InvalidParameter(
-                                    "Payment hash or amount mismatch".to_string(),
-                                ));
-                            }
-                            // TODO: check the expiry time, if expired, we should return an error.
+                let mut peeled_packet_bytes: Option<Vec<u8>> = None;
+
+                if !add_tlc.onion_packet.is_empty() {
+                    // TODO: Here we call network actor to peel the onion packet. Indeed, this message is forwarded from
```

Comment:
or we passing the `private_key` to channel state when creating channel actor, i'm not sure which way is better.